pkg/kubelet/dockertools/container_gc.go

```diff
@@ -245,35 +281,56 @@ func (cgc *containerGC) GarbageCollect(gcPolicy kubecontainer.ContainerGCPolicy,
 	return nil
 }
 
-func (cgc *containerGC) removeContainer(id string, podNameWithNamespace string, containerName string) {
-	glog.V(4).Infof("Removing container %q name %q", id, containerName)
-	err := cgc.client.RemoveContainer(id, dockertypes.ContainerRemoveOptions{RemoveVolumes: true})
+func (cgc *containerGC) netContainerCleanup(containerInfo containerGCInfo) {
```

Comment:
> unfortunately that kubecontainer.ConvertPodStatusToRunningPod() call only converts running containers, so the dead infra is left to GC and not cleaned up. Perhaps if we also solve this, it will fix the possible conflict with plugin workarounds by ensuring that the dead infra always gets cleaned up before a new one is started, as you suggest?

SyncPod only kills unwanted running containers, and all dead containers are left to GC. I'm concerned of introducing much sync latency if killing dead containers are added into syncPod.